### PR TITLE
starspace: fix build on aarch64-darwin

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/machine-learning/starspace/default.nix [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{ lib, stdenv, fetchFromGitHub, boost165, zlib }:
+{ lib, stdenv, fetchFromGitHub, boost, zlib }:
 
 stdenv.mkDerivation rec {
   pname = "starspace";
-  version = "unstable-2021-01-17";
+  version = "unstable-2019-12-13";
```
Comment (mausch) [inline pkgs/applications/science/machine-learning/starspace/default.nix:5]: Hmm 2021-01-17 was the date the package was created, not the date of the last commit of the repo. Very unfortunate. Not sure it's worth changing it?

Comment (wegank) [reply pkgs/applications/science/machine-learning/starspace/default.nix:5]: According to https://nixos.org/manual/nixpkgs/stable/#sec-package-naming:

> If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in "unstable-YYYY-MM-DD" format.

So I guess yes.

Comment (mausch) [reply pkgs/applications/science/machine-learning/starspace/default.nix:5]: Fair enough. Though I'd say put it in a separate commit at least?

Comment (wegank) [reply pkgs/applications/science/machine-learning/starspace/default.nix:5]: Done.

**Review by mausch — APPROVED** [review]

**Review by mausch — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**